sdk/inc/azure/storage/az_storage_blobs.h

```diff
@@ -33,16 +33,24 @@
  */
 static az_span const AZ_STORAGE_API_VERSION = AZ_SPAN_LITERAL_FROM_STR("2019-02-02");
 
+/**
+ * @brief Azure Storage Blobs blob client options
+ * @remark Allows customization of the blob client.
+ */
 typedef struct
 {
-  az_http_policy_retry_options retry;
+  az_http_policy_retry_options retry_options; /**< Optional values used to override the default retry policy options **/
   struct
   {
     _az_http_policy_apiversion_options api_version;
-    _az_http_policy_telemetry_options _telemetry_options;
+    _az_http_policy_telemetry_options telemetry_options;
   } _internal;
 } az_storage_blobs_blob_client_options;
 
+/**
+ * @brief Azure Storage Blobs Blob Client.
```

Comment:
Isn't that just kind-of-obvious? Can we add more details.

What is this struct used for? Any relevant quirks the user should know about?

Maybe:
```suggestion
 * @brief Client struct used to upload blobs to Azure Storage.
```

**What other languages say:**
> The BlobClient allows you to manipulate Azure Storage blobs.
https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.blobclient?view=azure-dotnet

> A client to interact with a specific blob, although that blob may not yet exist.
https://docs.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobclient?view=azure-python#download-blob-offset-none--length-none----kwargs-

I like the detail in the Java docs.
> This class provides a client that contains generic blob operations for Azure Storage Blobs. Operations allowed by the client are uploading and downloading, copying a blob, retrieving and setting metadata, retrieving and setting HTTP headers, and deleting and un-deleting a blob.
https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-blob/12.4.0/index.html